src/components/MaterialUI/DataDisplay/tableHelpers.test.js

```diff
@@ -0,0 +1,505 @@
+			{ id: "an_id2", test: 44, another: "another 2", extraneous: "Don't show 2" },
+			{ id: "an_id3", test: 0, another: "another 3", extraneous: "Don't show 3" },
+		];
+
+		const { headers, rows } = buildHeaderAndRowFromConfig(columnDef, elements);
+
+		expect(headers.length, "to equal", 1);
+
+		expect(rows.length, "to equal", 3);
+		expect(rows[0].columns.length, "to equal", 1);
+		expect(rows[0].element, "to equal", elements[0]);
+		expect(rows[0].columns[0].cellElement.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: 123,
+		});
+		expect(rows[0].columns[0].title.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: 123,
+		});
+
+		expect(rows[1].columns[0].cellElement.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: 44,
+		});
+		expect(rows[1].columns[0].title.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: 44,
+		});
+
+		expect(rows[2].columns[0].cellElement.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: "0",
+		});
+		expect(rows[2].columns[0].title.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: "0",
+		});
+	});
+
+	it("build table rows as expected with currency from elements", () => {
+		const columnDef = [
+			{
+				fieldName: "test",
+				label: messages.a_label,
+				type: "currency",
+				currency: ["currency"],
+			},
+		];
+		const elements = [
+			{
+				id: "an_id1",
+				test: 123,
+				another: "another 1",
+				extraneous: "Don't show 1",
+				currency: "EUR",
+			},
+			{
+				id: "an_id2",
+				test: 44,
+				another: "another 2",
+				extraneous: "Don't show 2",
+				currency: "USD",
+			},
+			{
+				id: "an_id3",
+				test: 0,
+				another: "another 3",
+				extraneous: "Don't show 3",
+				currency: "CAD",
+			},
+		];
+
+		const { headers, rows } = buildHeaderAndRowFromConfig(columnDef, elements);
+
+		expect(headers.length, "to equal", 1);
+
+		expect(rows.length, "to equal", 3);
+		expect(rows[0].columns.length, "to equal", 1);
+		expect(rows[0].element, "to equal", elements[0]);
+		expect(rows[0].columns[0].cellElement.props, "to equal", {
+			style: "currency",
+			currency: "EUR",
+			value: 123,
+		});
+		expect(rows[0].columns[0].title.props, "to equal", {
+			style: "currency",
+			currency: "EUR",
+			value: 123,
+		});
+
+		expect(rows[1].columns[0].cellElement.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: 44,
+		});
+		expect(rows[1].columns[0].title.props, "to equal", {
+			style: "currency",
+			currency: "USD",
+			value: 44,
+		});
+
+		expect(rows[2].columns[0].cellElement.props, "to equal", {
+			style: "currency",
+			currency: "CAD",
+			value: "0",
+		});
+		expect(rows[2].columns[0].title.props, "to equal", {
+			style: "currency",
+			currency: "CAD",
+			value: "0",
+		});
+	});
+
+	it("build table rows as expected with date", () => {
```

Comment:
I notice the data is numeric, but not a date?  Is this a valid test?

Reply:
Dates are actually numeric values and the purpose of the test is not actually to test the "FormattedDate" or the "FormattedTime" component from react-intl.